get-started/nodejs/build-images.md

```diff
@@ -0,0 +1,224 @@
+
+```shell
+$ cd [path to your node-docker directory]
+$ npm init -y
+$ npm install ronin-server ronin-mocks
+$ touch server.js
+```
+
+Now, let’s add some code to handle our REST requests. We’ll use a mock server so we can focus on Dockerizing the application.
+
+Open this working directory in your IDE and add the following code into the `server.js` file.
+
+```shell
+const ronin     = require( 'ronin-server' )
+const mocks     = require( 'ronin-mocks' )
+
+const server = ronin.server()
+
+server.use( '/', mocks.server( server.Router(), false, true ) )
+server.start()
+```
+
+The mocking server is called `Ronin.js` and will listen on port 8000 by default. You can make POST requests to the root (/) endpoint and any JSON structure you send to the server will be saved in memory. You can also send GET requests to the same endpoint and receive an array of JSON objects that you have previously POSTed.
+
+## Test application
+
+Let’s start our application and make sure it’s running properly. Open your terminal and navigate to your working directory you created.
+
+`$ node server.js`
+
+To test that the application is working properly, we’ll first POST some JSON to the API and then make a GET request to see that the data has been saved. Open a new terminal and run the following curl commands:
+
+```shell
+$ curl --request POST \
+  --url http://localhost:8000/test \
+  --header 'content-type: application/json' \
+  --data '{
+	"msg": "testing"
+}'
+{"code":"success","payload":[{"msg":"testing","id":"31f23305-f5d0-4b4f-a16f-6f4c8ec93cf1","createDate":"2020-08-28T21:53:07.157Z"}]}
+
+$ curl http://localhost:8000/test
+{"code":"success","meta":{"total":1,"count":1},"payload":[{"msg":"testing","id":"31f23305-f5d0-4b4f-a16f-6f4c8ec93cf1","createDate":"2020-08-28T21:53:07.157Z"}]}
+```
+
+Switch back to the terminal where our server is running. You should now see the following requests in the server logs.
+
+```shell
+2020-XX-31T16:35:08:4260  INFO: POST /test
+2020-XX-31T16:35:21:3560  INFO: GET /test
+```
+
+## Create a Dockerfile for Node.js
+
+A Dockerfile is a text document that contains all the commands a user could call on the command line to assemble an image. When we tell Docker to build our image by executing the `docker run` command, Docker reads these instructions and executes them one by one and creates a Docker image as a result.
+
+Let’s walk through the process of creating a Dockerfile for our application. In the root of your working directory, create a file named `Dockerfile` and open this file in your text editor.
+
+> **Note**
+>
+> The name of the Dockerfile is not important but the default filename for many commands is simply `Dockerfile`. So, we’ll use that as our filename throughout this series.
+
+The first thing we need to do is to add a line in our Dockerfile that tells Docker what base image we would like to use for our application.
+
+Dockerfile:
+`FROM node:12.18.1`
```

Comment:
I think it looks better when we use a block with three backticks for each line of the Dockerfile. and also with the ```dockerfile syntax highlighter

```dockerfile
FROM node:12.18
```

We could also remove the minor version number and just use `node:12.18` here, that reduces the effort to update the examples whenever there is a patch release of the base image.

All further places with `12.18.1` whould be updated as well.